File: solutions/scala/collatz-conjecture/1/src/main/scala/CollatzConjecture.scala
object CollatzConjecture{
  def steps(x:Int) : Option[Int] = turns(0,x)
  @scala.annotation.tailrec
  def turns(step:Int, cur:Int) : Option[Int] = if(cur>0) then (cur,cur%2) match{
    case (1,_) => Some(step)
    case (_,0) => turns(step+1, cur/2)
    case (_,1) => turns(step+1, (3*cur)+1)
  } else None

}

File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int roll){
    int i=0;
    if(roll<=0) return -1;
    while (roll>1){
        if(roll%2==0){
            roll=roll/2;
            i++;
        }else{
            roll=roll*3+1;
            i++;
        }
    }
    return i;
}